### ISSUE TITLE
BUG: Compability with Flow 5.2 (PdoBackend)

### ISSUE THREAD

**Issue by rolandschuetz** [flownative/beach-flow-companion#3]
When upgrading from neos/utility-pdo 5.1.5 to 5.2.1 it's not possible anymore to log into the backend.

This seems to be an issue with the PdoBackend class, see [stack trace](http://prntscr.com/lt038o) and logs:

```
Exception in line 90 of /Users/rolandschutz/Documents/Projects/Neos.DocsNeosIo/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Flownative_BeachFlowCompanion_Cache_PdoBackend.php: Call to undefined method Neos\Cache\Backend\PdoBackend::initializeObject()

38 Flownative\BeachFlowCompanion\Cache\PdoBackend_Original::initializeObject(1)
37 Flownative\BeachFlowCompanion\Cache\PdoBackend::__construct(Neos\Cache\EnvironmentConfiguration, array|1|)
36 Neos\Cache\CacheFactory::instantiateBackend("Flownative\BeachFlowCompanion\Cache\PdoBackend", array|1|, Neos\Cache\EnvironmentConfiguration)
35 Neos\Flow\Cache\CacheFactory::instantiateBackend("Flownative\BeachFlowCompanion\Cache\PdoBackend", array|1|, Neos\Cache\EnvironmentConfiguration, true)
34 Neos\Flow\Cache\CacheFactory::create("Flow_Security_Cryptography_HashService", "Neos\Cache\Frontend\StringFrontend", "Flownative\BeachFlowCompanion\Cache\PdoBackend", array|1|, true)
33 Neos\Flow\Cache\CacheManager::createCache("Flow_Security_Cryptography_HashService")
32 Neos\Flow\Cache\CacheManager::getCache("Flow_Security_Cryptography_HashService")
31 Neos\Flow\Security\Cryptography\HashService::Neos\Flow\Security\Cryptography\{closure}()
30 Closure::__invoke()
29 Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy::_activateDependency()
28 Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy::__call("get", array|1|)
27 Neos\Flow\Security\Cryptography\HashService_Original::getEncryptionKey()
26 Neos\Flow\Security\Cryptography\HashService_Original::generateHmac("YTowOnt9")
25 Neos\Flow\Security\Cryptography\HashService_Original::appendHmac("YTowOnt9")
24 call_user_func_array(array|2|, array|1|)
23 Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy::__call("appendHmac", array|1|)
22 Neos\FluidAdaptor\ViewHelpers\FormViewHelper_Original::renderHiddenReferrerFields()
21 Neos\FluidAdaptor\ViewHelpers\FormViewHelper_Original::render()
20 call_user_func_array(array|2|, array|0|)
19 Neos\FluidAdaptor\Core\ViewHelper\AbstractViewHelper::callRenderMethod()
18 TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper::initializeArgumentsAndRender()
17 TYPO3Fluid\Fluid\Core\ViewHelper\ViewHelperInvoker::invoke("Neos\FluidAdaptor\ViewHelpers\FormViewHelper", array|32|, Neos\FluidAdaptor\Core\Rendering\RenderingContext, Closure)
16 TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper::renderStatic(array|32|, Closure, Neos\FluidAdaptor\Core\Rendering\RenderingContext)
15 Login_action_index_637ac4ff24b456a5dbf24abf29f2fa64728618c5::section_02083f4579e08a612425c0c1a17ee47add783b94(Neos\FluidAdaptor\Core\Rendering\RenderingContext)
14 TYPO3Fluid\Fluid\View\AbstractTemplateView::renderSection("body", array|4|, false)
13 Neos\FluidAdaptor\View\AbstractTemplateView::renderSection("body", array|4|, false)
12 TYPO3Fluid\Fluid\ViewHelpers\RenderViewHelper::renderStatic(array|8|, Closure, Neos\FluidAdaptor\Core\Rendering\RenderingContext)
11 layout_Default_html_cad78be3eb58535f50d7adf3fba58cd089d5299b::render(Neos\FluidAdaptor\Core\Rendering\RenderingContext)
10 TYPO3Fluid\Fluid\View\AbstractTemplateView::render()
9 Neos\Flow\Mvc\Controller\ActionController_Original::callActionMethod()
8 Neos\Flow\Mvc\Controller\ActionController_Original::processRequest(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
7 Neos\Flow\Mvc\Dispatcher_Original::initiateDispatchLoop(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
6 Neos\Flow\Mvc\Dispatcher_Original::dispatch(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
5 Neos\Flow\Mvc\DispatchComponent_Original::handle(Neos\Flow\Http\Component\ComponentContext)
4 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
3 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
2 Neos\Flow\Http\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()


HTTP REQUEST:
GET /neos/login HTTP/1.1
Host: 127.0.0.1:8081
Cache-Control: no-cache
Connection: keep-alive
Pragma: no-cache
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36
Dnt: 1
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: de-AT,de;q=0.9,en-AT;q=0.8,en;q=0.7,de-DE;q=0.6,en-US;q=0.5



HTTP RESPONSE:
[response was empty]

PHP PROCESS:
Inode: 116000304
PID: 60725
UID: 501
GID: 20
User: rolandschutz


```

**Comment by kitsunet** [comment]
In fact the initializeObject of the parent PdoBackend doesn't exist anymore and `initializeObject` shouldn't have been called since the `\Neos\Flow\Cache\Backend\FlowSpecificBackendInterface` was removed.

See also `\Neos\Flow\Cache\CacheFactory::instantiateFlowSpecificBackend` which should actually be removed as a bugfix in >= 5.0 because the respective condition in 151 of that class will never be met (as the interface is gone).

**Comment by rolandschuetz** [comment]
Ok, so what should I do. Every time I load neos/utility-pdo 5.1.5 via composer the backend works, when I switch to 5.2.1 and delete all caches the backend is not usable. I'm not using any custom PHP here, so that needs to be fixed in this package.

**Comment by rolandschuetz** [comment]
@kitsunet Could this be because of this setting?
https://www.flownative.com/en/documentation/guides/beach/how-to-store-google-analytics-tokens-in-datebase.html

**Comment by kdambekalns** [comment]
Released with 1.0.6
